packages/react-utils/src/helpers/dataLoaders.ts

```diff
@@ -33,6 +37,86 @@ export class OpenSRPService<T extends object = Dictionary> extends GenericOpenSR
   }
 }
 
+/** A generic FHIR service class
+ *
+ * Sample usage:
+ * -------------
+ * const serve = new FHIRServiceClass('<base url>', '<resource type>');
+ *
+ * **To list all entries of a resource (GET request)**: serve.list()
+ *
+ * **To get one resource record**: service.read('<object id>')
+ *
+ * **To create a new resource**: service.create(payload)
+ *
+ * **To update a resource record**: service.update(payload)
+ */
+export class FHIRServiceClass<T = fhirclient.FHIR.Resource> {
+  public accessTokenOrCallBack: GetAccessTokenType | string;
+  public baseURL: string;
+  public resourceType: string;
+
+  /**
+   * Constructor method
+   *
+   * @param {string} baseURL - the base FHIR URL
+   * @param {string} resourceType - FHIR resource type string
+   */
+  constructor(baseURL: string, resourceType: string) {
+    this.accessTokenOrCallBack = handleSessionOrTokenExpiry;
+    this.baseURL = baseURL;
+    this.resourceType = resourceType;
+  }
+
+  public buildQueryParams(params: URLParams | null) {
+    if (params) {
+      return `${this.resourceType}/_search?${decodeURIComponent(queryString.stringify(params))}`;
+    }
+    return this.resourceType;
+  }
```

Comment:
Do we really need this implementation i have already implemented it in Server side pagination.  lets not implement this here and reduce complexity of PR ?

Implementation :  https://github.com/opensrp/web/blob/164aaf0c12f71f72117698ac85fb5b4304babb94/packages/react-utils/src/components/PaginateData/index.tsx#L63-L68
Working : https://github.com/opensrp/web/blob/164aaf0c12f71f72117698ac85fb5b4304babb94/packages/react-utils/src/components/PaginateData/tests/index.test.tsx#L108-L147